pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -24,98 +22,61 @@ var mercuryInfo = capabilities.MustNewCapabilityInfo(
 	"v1.0.0",
 )
 
+const defaultTickerResolutionMs = 1000
+
 // This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
 type MercuryTriggerService struct {
 	capabilities.CapabilityInfo
-	chans               map[string]chan<- capabilities.CapabilityResponse
-	feedIDsForTriggerID map[string][]mercury.FeedID
-	triggerIDsForFeedID map[mercury.FeedID]map[string]bool
-	mu                  sync.Mutex
-	lggr                logger.Logger
+	tickerResolutionMs int64
+	chans              map[string]chan<- capabilities.CapabilityResponse
+	subscribers        map[string]*subscriber
+	latestReports      map[mercury.FeedID]mercury.FeedReport
+	mu                 sync.Mutex
+	stopCh             services.StopChan
+	wg                 sync.WaitGroup
+	lggr               logger.Logger
+}
+
+var _ services.Service = &MercuryTriggerService{}
+
+type SubscriberConfig struct {
```

Comment:
Does this need to be public?

Reply:
Ignore: I assume there's a follow up PR that will use this 👍

Reply:
no, changing